### DISCUSSION TITLE
Configure Scope in Transaction/Request

### DISCUSSION THREAD

**Discussion by billypap1** [getsentry/sentry-go#814] [Q&A] [answered]
I use Sentry in my Go App - Gin  and I set scope in a middleware. 
Before each route I set user details (via the middleware): 

```
sentry.ConfigureScope(func(scope *sentry.Scope) {
	scope.SetUser(sentry.User{Email: "jane.doe@example.com"})
})
```

ConfigureScope is configured globally or per each transaction/request that Gin handles?

**Comment by cleptric** [comment]
https://docs.sentry.io/platforms/go/usage/concurrency/

You might end up with something we call scope bleed by using the global function; hence I recommend something like this in your middleware:

```go
hub := sentry.GetHubFromContext(ctx)
if hub != nil {
    hub.ConfigureScope(func(scope *sentry.Scope) {
	scope.SetUser(sentry.User{Email: "jane.doe@example.com"})
    })
}
```

**Reply by billypap1** [reply]
@cleptric I want to set the user only for this specific request/transaction. Should I set it in hub?

**Reply by cleptric** [reply, marked as answer]
Yes, the hub is cloned on each request, and the correct scoping is guaranteed per request.
See https://github.com/getsentry/sentry-go/blob/master/gin/sentrygin.go#L54-L59

**Reply by billypap1** [reply]
@cleptric Ok, thank you!
